modules/gateway/upnp.go

```diff
@@ -38,10 +38,15 @@ func myExternalIP() (string, error) {
 	return string(buf[:n-1]), nil
 }
 
-// learnHostname discovers the external IP of the Gateway. Once the IP has
-// been discovered, it registers the ShareNodes RPC to be called on new
+// threadedLearnHostname discovers the external IP of the Gateway. Once the IP
+// has been discovered, it registers the ShareNodes RPC to be called on new
 // connections, advertising the IP to other nodes.
-func (g *Gateway) learnHostname(port string) {
+func (g *Gateway) threadedLearnHostname() {
```

Comment:
I think I slightly prefer the current version, since it's a little strange for a function named `learnHostname` to take a port as an argument.